droughtpipeline/load.py

```diff
@@ -0,0 +1,872 @@
+import requests
+import geopandas as gpd
+from typing import List
+import shutil
+from azure.storage.blob import BlobServiceClient
+from azure.core.exceptions import ResourceNotFoundError
+
+COSMOS_DATA_TYPES = [
+    "climate-region",
+    "seasonal-rainfall-forecast",
+]
+
+
+def get_cosmos_query(
+    start_date=None,
+    end_date=None,
+    country=None,
+    adm_level=None,
+    climate_region_code=None,
+    pcode=None,
+    lead_time=None,
+):
+    query = "SELECT * FROM c WHERE "
+    if start_date is not None:
+        query += f'c.timestamp >= "{start_date.strftime("%Y-%m-%dT%H:%M:%S")}" '
+    if end_date is not None:
+        query += f'AND c.timestamp <= "{end_date.strftime("%Y-%m-%dT%H:%M:%S")}" '
+    if country is not None:
+        query += f'AND c.country = "{country}" '
+    if adm_level is not None:    
+        query += f'AND c.adm_level = "{adm_level}" '
+    if climate_region_code is not None:    
+        query += f'AND c.climate_region_code = "{climate_region_code}" '
+    if pcode is not None:
+        query += f'AND c.adm_level = "{pcode}" '
+    if lead_time is not None:
+        query += f'AND c.adm_level = "{lead_time}" '
+    if query.endswith("WHERE "):
+        query = query.replace("WHERE ", "")
+    query = query.replace("WHERE AND", "WHERE")
+    return query
+
+
+def get_data_unit_id(data_unit: AdminDataUnit, dataset: AdminDataSet):
+    """Get data unit ID"""
+    if hasattr(data_unit, "pcode"):
+        if hasattr(data_unit, "lead_time"):
+            id_ = f"{data_unit.pcode}_{dataset.timestamp.strftime('%Y-%m-%dT%H:%M:%S')}_{data_unit.lead_time}"
+        else:
+            id_ = f"{data_unit.pcode}_{dataset.timestamp.strftime('%Y-%m-%dT%H:%M:%S')}"
+    elif hasattr(data_unit, "climate_region_code"):
+        if hasattr(data_unit, "lead_time"):
+            id_ = f"{data_unit.climate_region_code}_{dataset.timestamp.strftime('%Y-%m-%dT%H:%M:%S')}_{data_unit.lead_time}"
+        else:
+            id_ = f"{data_unit.climate_region_code}_{dataset.timestamp.strftime('%Y-%m-%dT%H:%M:%S')}"
+    else:
+        id_ = f"{dataset.timestamp.strftime('%Y-%m-%dT%H:%M:%S')}"
+    return id_
+
+
+def alert_class_to_threshold(alert_class: str, triggered: bool) -> float:
+    """Convert alert class to 'alert_threshold'"""
+    if alert_class == "no":
+        return 0.0
+    elif alert_class == "min":
+        return 0.3
+    elif alert_class == "med":
+        return 0.7
+    elif alert_class == "max":
+        if triggered:
+            return 1.0
+        else:
+            return 0.7
+    else:
+        raise ValueError(f"Invalid alert class {alert_class}")
+    
+def forecast_trigger_status(triggered: bool, trigger_class: str) -> float:
+    """determine if forecast is a trigger for IBF portal"""
+    if triggered:
+        if trigger_class == "disabled":
+            return 0
+        else:
+            return 1
+    else:
+        raise ValueError(f"Invalid alert class {triggered}")
```

Comment:
I don't get this 'else'. This says that if triggered=false is passed, it returns an error? It seems like you tied this 'else' to the wrong 'if', it should instead be tied to the if on 'trigger_class'?

Reply:
good catch thanks!
def forecast_trigger_status(triggered: bool, trigger_class: str) -> int:
    """determine if forecast is a trigger for IBF portal
        if trigger status is true and trigger activation is enabled in config file the 
        trigger staus will be 1 , else 0"""
    if triggered:
        if trigger_class == "enabled":
            return 1
    else:
        return 0